src/Razor/src/Microsoft.NET.Sdk.Razor/build/netstandard2.0/Microsoft.NET.Sdk.Razor.MvcApplicationPartsDiscovery.targets

```diff
@@ -0,0 +1,103 @@
+<!--
+***********************************************************************************************
+Microsoft.NET.Sdk.Razor.ApplicationPartsDiscovery
+
+WARNING:  DO NOT MODIFY this file unless you are knowledgeable about MSBuild and have
+          created a backup copy.  Incorrect changes to this file will make it
+          impossible to load or build your projects from the command-line or the IDE.
+
+Copyright (c) .NET Foundation. All rights reserved.
+***********************************************************************************************
+-->
+
+<Project ToolsVersion="14.0">
+  <UsingTask
+    TaskName="Microsoft.AspNetCore.Razor.Tasks.DiscoverMvcApplicationParts"
+    AssemblyFile="$(RazorSdkBuildTasksAssembly)"
+    Condition="'$(RazorSdkBuildTasksAssembly)' != ''" />
+
+  <PropertyGroup>
+    <GenerateMvcApplicationPartsAttribute Condition="'$(GenerateMvcApplicationPartsAssemblyAttributes)' == '' AND '$(OutputType)' == 'Exe'">true</GenerateMvcApplicationPartsAttribute>
+
+    <CoreCompileDependsOn Condition="'$(GenerateMvcApplicationPartsAttribute)' == 'true' AND '$(DesignTimeBuild)' != 'true'">
+      _DiscoverMvcApplicationParts;
+      $(CoreCompileDependsOn);
+    </CoreCompileDependsOn>
+
+    <_MvcApplicationPartAttributeGeneratedFile>$(IntermediateOutputPath)$(TargetName).MvcApplicationPartsAssemblyInfo$(DefaultLanguageSourceExtension)</_MvcApplicationPartAttributeGeneratedFile>
+
+    <_DiscoverMvcApplicationPartsDependsOn>
+      ResolveAssemblyReferences;
+      _InitializeDiscoverMvcApplicationParts;
+    </_DiscoverMvcApplicationPartsDependsOn>
```

Comment:
Do we need this chaining here? This could just be written in the target definition.

Reply:
Agreed, since it's underscored ("private") people shouldn't be customizing using it, so no need to put it in a property.